include/envoy/registry/registry.h

```diff
@@ -23,27 +24,37 @@ template <class Base> class FactoryRegistry;
 template <class T, class Base> class RegisterFactory;
 
 /**
- * Helper class to call `registeredNames` for a specialized
- * FactoryRegistry.
+ * FactoryRegistryProxy is a proxy object that provides access to the
+ * static methods of a strongly-typed factory registry.
  */
-class BaseFactoryCategoryNames {
+class FactoryRegistryProxy {
 public:
-  virtual ~BaseFactoryCategoryNames() = default;
+  virtual ~FactoryRegistryProxy() = default;
   virtual std::vector<absl::string_view> registeredNames() const PURE;
+  virtual bool disableFactory(absl::string_view) PURE;
 };
 
-template <class Base> class FactoryCategoryNames : public BaseFactoryCategoryNames {
+template <class Base> class FactoryRegistryProxyImpl : public FactoryRegistryProxy {
 public:
   using FactoryRegistry = Envoy::Registry::FactoryRegistry<Base>;
 
   std::vector<absl::string_view> registeredNames() const override {
     return FactoryRegistry::registeredNames();
   }
+
+  bool disableFactory(absl::string_view name) override {
+    return FactoryRegistry::disableFactory(name);
+  }
 };
 
+/**
+ * BaseFactoryCategoryRegistry holds the static factory map for
+ * FactoryCategoryRegistry, ensuring that friends of that class
+ * cannot get non-const access to it.
+ */
 class BaseFactoryCategoryRegistry {
 protected:
-  using MapType = std::map<std::string, BaseFactoryCategoryNames*>;
+  using MapType = std::map<std::string, FactoryRegistryProxy*, std::less<>>;
```

Comment:
redundant `std::less<>`

Reply:
This is needed so that we can index the map with `string_view` even though it has `string` keys (heterogenous lookup).

Reply:
Fixed.